client/proto/CMakeLists.txt

```diff
@@ -1,21 +1,105 @@
-find_package(Protobuf REQUIRED)
+cmake_minimum_required(VERSION 3.5)
+set(CMAKE_CXX_STANDARD 17)
+
+project(concord)
+
+find_package(absl CONFIG REQUIRED)
 find_package(GRPC REQUIRED)
 
-include_directories(${GRPC_INCLUDE_DIR})
-
-protobuf_generate_cpp(PROTO_SRCS PROTO_HDRS ${CMAKE_CURRENT_BINARY_DIR}
-  request/v1/request.proto
-  event/v1/event.proto
-  state_snapshot/v1/state_snapshot.proto
-  ../concordclient/proto/concord_client_request/v1/concord_client_request.proto
-)
-grpc_generate_cpp(GRPC_SRCS GRPC_HDRS ${CMAKE_CURRENT_BINARY_DIR}
-  request/v1/request.proto
-  event/v1/event.proto
-  state_snapshot/v1/state_snapshot.proto
-  ../concordclient/proto/concord_client_request/v1/concord_client_request.proto
-)
-
-add_library(clientservice-proto STATIC ${PROTO_SRCS} ${GRPC_SRCS})
-target_link_libraries(clientservice-proto PRIVATE protobuf::libprotobuf gRPC::grpc++)
+# Proto file
+get_filename_component(rs_proto "request/v1/request.proto" ABSOLUTE)
+get_filename_component(es_proto "event/v1/event.proto" ABSOLUTE)
+get_filename_component(ss_proto "state_snapshot/v1/state_snapshot.proto" ABSOLUTE)
+get_filename_component(ccr_proto "../concordclient/proto/concord_client_request/v1/concord_client_request.proto" ABSOLUTE)
+get_filename_component(rs_proto_path "${rs_proto}" PATH)
+get_filename_component(es_proto_path "${es_proto}" PATH)
+get_filename_component(ss_proto_path "${ss_proto}" PATH)
+get_filename_component(ccr_proto_path "${ccr_proto}" PATH)
+
+get_target_property(grpc_cpp_plugin_location gRPC::grpc_cpp_plugin LOCATION)
+
+# Generated sources
+set(rs_proto_srcs "${CMAKE_CURRENT_BINARY_DIR}/request.pb.cc")
```

Comment:
Why can't we use `grpc_generate_cpp()` any more ?

Reply:
@vrancurel the [FindProtobuf.cmake](https://github.com/Kitware/CMake/blob/master/Modules/FindProtobuf.cmake#L103) file specifies that [protobuf_generate_cpp works properly ](https://github.com/Kitware/CMake/blob/01e95efc34a2a9bcf9910d65d3b38ff9b8a51750/Modules/FindProtobuf.cmake#L103)within the same directory. I am not entirely sure what it means, but with our directory structure, I wasn’t able to compile the proto files using the functions `protobuf_generate_cpp` and `grpc_generate_cpp`.
Currently all the examples in the grpc cpp repo, also compile the proto files using the cmake custom command (for e.g., [this](https://github.com/grpc/grpc/blob/master/examples/cpp/helloworld/CMakeLists.txt))